packages/compass-collection/src/components/collection-header/collection-header.tsx

```diff
@@ -170,8 +187,40 @@ export const CollectionHeader: React.FunctionComponent<
           namespace={namespace}
           sourceName={sourceName}
           sourcePipeline={sourcePipeline}
+          onOpenMockDataModal={onOpenMockDataModal}
         />
       </div>
+      <MockDataGeneratorModal
+        isOpen={isMockDataModalOpen}
+        onClose={onCloseMockDataModal}
+        currentStep={currentStep}
+        onCurrentStepChange={onSetMockDataGeneratorStep}
+      />
     </div>
   );
 };
+
+const mapStateToProps = (state: CollectionState) => ({
+  isMockDataModalOpen: state.mockDataGenerator.isModalOpen,
+  currentStep: state.mockDataGenerator.currentStep,
+});
+
+type ConnectedCollectionHeaderProps = Omit<
+  CollectionHeaderProps,
+  | 'isMockDataModalOpen'
+  | 'currentStep'
+  | 'onOpenMockDataModal'
+  | 'onCloseMockDataModal'
+  | 'onSetMockDataGeneratorStep'
+>;
+
+const ConnectedCollectionHeader = connect(mapStateToProps, {
+  onOpenMockDataModal: mockDataGeneratorModalOpened,
+  onCloseMockDataModal: mockDataGeneratorModalClosed,
+  onSetMockDataGeneratorStep: mockDataGeneratorStepChanged,
+})(CollectionHeader) as React.ComponentType<ConnectedCollectionHeaderProps>;
```

Comment:
Redux connect is pretty good at correctly deriving the types for connected components, usually you don't need to assert types unless your types are incorrect, can you clarify why you are doing an assertion here?